clang/lib/Tooling/DependencyScanning/ScanAndUpdateArgs.cpp

```diff
@@ -26,11 +26,19 @@ void tooling::dependencies::configureInvocationForCaching(
   auto &FrontendOpts = CI.getFrontendOpts();
   FrontendOpts.CacheCompileJob = true;
   FrontendOpts.IncludeTimestamps = false;
+
   // Clear this otherwise it defeats the purpose of making the compilation key
   // independent of certain arguments.
-  CI.getCodeGenOpts().DwarfDebugFlags.clear();
+  auto &CodeGenOpts = CI.getCodeGenOpts();
+  if (CI.getFrontendOpts().ProgramAction != frontend::ActionKind::EmitObj) {
+    CodeGenOpts.UseCASBackend = false;
```

Comment:
I missed the original PR but a bit simpler change would be to clear these options unconditionally inside the `resetBenignCodeGenOptions()` function.

Reply:
But this is fine, not necessary to change.

Reply:
Sorry I forgot to add you to the PR for the change in next 😅

Reply:
Are these options actually "benign" though? My understanding is they're actively broken when building AST files.  That's different from the options we're clearing to get cache hits.

Reply:
Fair point, maybe such a change would be a misuse of that function.

Reply:
> Are these options actually "benign" though? My understanding is they're actively broken when building AST files. 

On second thought, conceptually they *are* benign, in that the presence of these flags for a PCH/PCM invocation does not affect the output. The fact that they cause a problem in the frontend execution is an implementation detail, they are not supposed to have an effect.

Clearing them in `resetBenignCodeGenOptions()` is better because it also removes them from the module context.

Reply:
> the presence of these flags for a PCH/PCM invocation does not affect the output

This is only because we don't implement them for modules.  If we supported writing pcm/pch debug info in cas friendly format they would not be benign, and we would need to preserve them.  From my perspective they're not benign, we just haven't implemented them.

Reply:
`DebugTypeExtRefs` is a similar case.  It "doesn't affect output", but only because we disable it because it's not supported yet for caching outside of Swift.

Reply:
> This is only because we don't implement them for modules. If we supported writing pcm/pch debug info in cas friendly format they would not be benign, and we would need to preserve them

Ok, fair point, but then why do we remove them in the first place, let them through and handle them in the PCM frontend invocation ("handle them" could be "ignore them for now; but later on they could restructure PCM's debug info"). What is the code change for this?

Reply:
That would also be fine with me. I'm guessing the current patch is simpler than handling it later during the invocation, but I could be convinced.

Reply:
@rastogishubham could you take a look and see what the change would look like for ignoring the flags in the PCM/PCH frontend invocation?